test/boost/sstable_compaction_test.cc

```diff
@@ -5209,3 +5209,22 @@ SEASTAR_TEST_CASE(test_large_partition_splitting_on_compaction) {
 
     });
 }
+
+// Without commit aba475fe1d24d5c, scylla will fail miserably (either with abort or segfault; depends on the version).
```

Comment:
I'm a bit lost :) Why without aba475fe1d24d5c, but not without "this" commit? AFAICS abort then drain then stop leads to bad state in master and this commit fixes it

Reply:
This conditional (https://github.com/scylladb/scylladb/commit/aba475fe1d24d5c2581bc5763f40924bbf688500#diff-8391e192323b62aa6c8460caaa8b772232e26d14c8bfeaeba2dc33ec7d69207bR511) is true when abort source subscription is called. It's one-off.

So your patch fixed the problem when compaction_manager::drain() (as a result of drain initiated via API for example) is called during shutdown, after do_stop() (from abort source subscription) and before stop() (from main's deferred shutdown)

Reply:
I was surprised too :-)

Reply:
Ah, indeed. Once abort source fires the subscription invalidates